bigquery_etl/pytest_plugin/sql.py

```diff
@@ -107,11 +107,12 @@ def runtest(self):
                 views[table_name] = read(self.fspath.strpath, resource)
 
         # rewrite all udfs as temporary
-        temp_udfs = parse_udf.sub_persisent_udfs_as_temp(query)
+        temp_udfs = parse_udf.sub_persistent_udfs_as_temp(query)
         if temp_udfs != query:
-            query = temp_udfs
             # prepend udf definitions
             query = parse_udf.prepend_udf_usage_definitions(query)
+            query = parse_udf.sub_persistent_udfs_as_temp(query)
+            query = query.replace("CREATE OR REPLACE FUNCTION", "CREATE TEMP FUNCTION")
```

Comment:
we should extract this logic to a function that we also call from `udf_tests_sql`, rather than duplicating it